dev-tools/kubernetes/base/elastic-agent-managed/daemonset.yaml

```diff
@@ -0,0 +1,116 @@
+metadata:
+  name: daemonset
+spec:
+  template:
+    spec:
+      # Tolerations are needed to run Elastic Agent on Kubernetes control-plane nodes.
+      # Agents running on control-plane nodes collect metrics from the control plane components (scheduler, controller manager) of Kubernetes
+      tolerations:
+        - key: node-role.kubernetes.io/control-plane
+          effect: NoSchedule
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+      serviceAccountName: service-account
+      hostNetwork: true
+      # 'hostPID: true' enables the Elastic Security integration to observe all process exec events on the host.
+      # Sharing the host process ID namespace gives visibility of all processes running on the same host.
+      hostPID: true
+      dnsPolicy: ClusterFirstWithHostNet
+      containers:
+        - name: elastic-agent
+          image: docker.elastic.co/beats/elastic-agent:8.1.0
+          env:
+            # Set to 1 for enrollment into Fleet server. If not set, Elastic Agent is run in standalone mode
+            - name: FLEET_ENROLL
+              value: "1"
+            - name: NODE_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: spec.nodeName
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+          envFrom:
+          - configMapRef:
+              # Fleet Server URL to enroll the Elastic Agent into
+              # FLEET_URL can be found in Kibana, go to Management > Fleet > Settings
+              # - name: FLEET_URL
+              #   value: "https://fleet-server:8220"
+              # Elasticsearch API key used to enroll Elastic Agents in Fleet (https://www.elastic.co/guide/en/fleet/current/fleet-enrollment-tokens.html#fleet-enrollment-tokens)
+              # If FLEET_ENROLLMENT_TOKEN is empty then KIBANA_HOST, KIBANA_FLEET_USERNAME, KIBANA_FLEET_PASSWORD are needed
+              # - name: FLEET_ENROLLMENT_TOKEN
+              #   value: "token-id"
+              # Set to true to communicate with Fleet with either insecure HTTP or unverified HTTPS
+              # - name: FLEET_INSECURE
+              #   value: "true"
+              # - name: KIBANA_HOST
+              #   value: "http://kibana:5601"
+              # # The basic authentication username used to connect to Kibana and retrieve a service_token to enable Fleet
+              # - name: KIBANA_FLEET_USERNAME
+              #   value: "elastic"
+              # # The basic authentication password used to connect to Kibana and retrieve a service_token to enable Fleet
+              # - name: KIBANA_FLEET_PASSWORD
+              #   value: "changeme"
+              name: configs
+          securityContext:
+            runAsUser: 0
+          resources:
+            limits:
+              memory: 500Mi
+            requests:
+              cpu: 100m
+              memory: 200Mi
+          volumeMounts:
```

Comment:
I think following part can also be splitted to multilple overalys. 

We want to have needed mounts for observability and extra ones that imposed from security

Reply:
do you mind elaborate a bit more on the topic. I vaguely recollect some suggestions from the security team but I wouldn't know how to change this part of the manifest